ebpf_extensions/neteventebpfext/netevent_ebpf_ext_event.c

```diff
@@ -237,6 +237,18 @@ _netevent_ebpf_extension_netevent_on_client_attach(
 
     UNREFERENCED_PARAMETER(provider_context);
 
+    if (client_data == NULL ||
```

Comment:
note to reviewers: It looks like the existing code may have (by design?) allowed for no attach params to be passed. I have changed this behavior to enforce that the caller always pass attach params, with this code enforcing that. Please let me know if anyone sees issue with this.

Reply:
Seems fine to me. With these new checks, you can remove the now redundant checks below in the existing code